### PR TITLE
popupdialog associated with an id (query selector)

### LINKED ISSUE #2293: Have part or dialog centered to the element clicked
### Description

Here, I have clicked on an icon and I have a dropdown menu of labels next to where I clicked:
image

Here, I have clicked on icon and I see a dialog/part showing up next to where I clicked:

image

I want to do that generically to put anything in this part. If I click somewhere else, this dialog should disappear. 

### Acceptance Criteria

- [ ] If applicable, a new demo code is provided to show the new feature in action.
- [ ] Integration tests exhibiting how the functionality works are added.
- [ ] Any new code is covered by a unit tested.
- [ ] Check code coverage is at least 90%.
- [ ] Related issue(s) in taipy-doc are created for documentation and Release Notes are updated.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
- popup dialog associated with an id (query selector)
- also fix a possible bug in chart

## Related Tickets & Documents

- Closes #2293 

## How to reproduce the issue
```py
from taipy.gui import Gui, State

ref_id = None
pop_open = False
val = None

def on_action(state: State, id: str):
  if id == "dialog":
    state.ref_id = None
    state.pop_open = False
  elif id:
    state.ref_id = f"#{id}"
    state.pop_open = True
  else:
      state.val = state.val +1 if state.val is not None else 1

page = """
<|button 1 |button|id=id1|>
<|button 2|button|id=id2|>


<|button |button|id=id3|>
<|button |button|id=id4|>
<|button |button|id=id5|>


<|button |button|id=id6|>
<|button |button|id=id7|>
<|button |button|id=id8|>
<|button |button|id=id9|>


<|button |button|id=id10|>
<|button |button|id=id11|>

<|{pop_open}|dialog|popup|ref_id={ref_id}|id=dialog|
<|inside the popup|button|>

<|{val}|>
|>
"""

if __name__ == "__main__":
  Gui(page).run(title="2293 popup")

```

### CODE REVIEW

File: taipy/gui/_renderers/factory.py [hunk 1]
```diff
@@ -215,6 +215,8 @@ class _Factory:
                 ("width", PropertyType.string_or_number),
                 ("height", PropertyType.string_or_number),
                 ("hover_text", PropertyType.dynamic_string),
+                ("ref_id", PropertyType.dynamic_string),
```
Comment (FabienLelaquais) [inline taipy/gui/_renderers/factory.py:218]:
```suggestion
                ("anchor_id", PropertyType.dynamic_string),
```
?
Do you see a use case for dynamicity?

Comment (FabienLelaquais) [reply taipy/gui/_renderers/factory.py:218]: Well your demo shows how useful this can be.
I'm in!

Comment (FabienLelaquais) [inline taipy/gui/_renderers/factory.py:218]: I suspect we will want an anchor_type (??) as well: center/below/...

File: frontend/taipy-gui/src/components/Taipy/Dialog.tsx [hunk 2]
```diff
@@ -110,7 +139,22 @@ const Dialog = (props: DialogProps) => {
...
+
+    return popup ? (
+        <Popover
+            id={id}
+            onClose={handleAction}
+            open={open === undefined ? defaultOpen === "true" || defaultOpen === true : !!open}
+            className={`${className} ${getComponentClassName(props.children)}`}
+            sx={paperProps.sx}
+            anchorEl={getAnchorEl}
+            anchorOrigin={popoverAnchor}
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Dialog.tsx:152]: This we will want to customize...

**Review by FabienLelaquais — APPROVED** [review]
👍 simple enough!
A little more work for popup placement.

File: taipy/gui/_renderers/factory.py (hunk 1 above)
Comment (FredLL-Avaiga) [reply taipy/gui/_renderers/factory.py:218]: not sure

File: frontend/taipy-gui/src/components/Taipy/Dialog.tsx (hunk 2 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Dialog.tsx:152]: not very interesting, we're positioning the popup at the center of the HTML element whose selector is passed

Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Dialog.tsx:152]: Well... an objective is to use this to mimic drop-down menus...

**Pull request merged**